### PR TITLE
image-builder: use storage/images-repos/{imageID} end-point in image-builder requests.

### PR DESCRIPTION
**Pull request by ldjebran**
Fixes THEEDGE-2730.

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by mgold1234 — APPROVED** [review]
/lgtm

File: pkg/services/images.go [hunk 1]
```diff
@@ -340,15 +340,23 @@ func (s *ImageService) UpdateImage(image *models.Image, previousImage *models.Im
...
-		repo, err := s.RepoService.GetRepoByID(previousSuccessfulImage.Commit.RepoID)
-		if err != nil {
-			s.log.WithField("error", err.Error()).Error("Commit repo wasn't found on the database")
-			err := errors.NewBadRequest(fmt.Sprintf("Commit repo wasn't found in the database: #%v", image.Commit.ID))
-			return err
+		if feature.StorageImagesRepos.IsEnabled() {
+			image.Commit.OSTreeParentCommit = fmt.Sprintf(
+				"%s/api/edge/v1/storage/images-repos/%d",
+				config.Get().EdgeCertAPIBaseURL,
+				previousSuccessfulImage.ID,
```
Comment (omaciel) [inline pkg/services/images.go:347]: Should this be `previousSuccessfulImage.Commit.RepoID`?

File: pkg/clients/imagebuilder/client_test.go [hunk 2]
```diff
@@ -351,6 +351,108 @@ var _ = Describe("Image Builder Client Test", func() {
 		Expect(img).ToNot(BeNil())
 		Expect(img.Commit.ComposeJobID).To(Equal("compose-job-id-returned-from-image-builder"))
 	})
+
+	Context("edge-management.storage_images_repos  feature", func() {
```
Comment (omaciel) [inline pkg/clients/imagebuilder/client_test.go:355]: How about negative tests?

File: pkg/services/images_test.go [hunk 3]
```diff
@@ -732,6 +733,55 @@ var _ = Describe("Image Service Test", func() {
 				Expect(image.Commit.OSTreeRef).To(Equal("rhel/9/x86_64/edge"))
 			})
 		})
+		Context("edge-management.storage_images_repos feature", func() {
```
Comment (omaciel) [inline pkg/services/images_test.go:736]: are negative tests implicit in the existing tests?

**Review by omaciel — COMMENTED** [review]
I haven't checked the code coverage for all functions that were touched :)

Leaving a comment for now.

File: pkg/services/images.go (hunk 1 above)
Comment (ldjebran) [reply pkg/services/images.go:347]: no no this has to be the image id

Comment (omaciel) [reply pkg/services/images.go:347]: Thank you. Marking it as resolved.

File: pkg/services/images_test.go (hunk 3 above)
Comment (ldjebran) [reply pkg/services/images_test.go:736]: same comment here

File: pkg/clients/imagebuilder/client_test.go (hunk 2 above)
Comment (ldjebran) [reply pkg/clients/imagebuilder/client_test.go:355]: eg all other tests in this module are considered feature disabled . when the feature will be permanent , we should delete this tests

Comment (omaciel) [reply pkg/clients/imagebuilder/client_test.go:355]: So implicit negative test. Thank you. Marking it as resolved.

**Review by omaciel — APPROVED** [review]
/lgtm

**Pull request merged**